compiler/parser/src/fstring.rs

```diff
@@ -194,21 +202,21 @@ impl FStringParser {
                             }),
                         ]
                     };
-                    return Ok((ret, chars));
+                    return Ok(ret);
                 }
                 '"' | '\'' => {
                     expression.push(ch);
-                    let mut string_ended = false;
-                    for next in &mut chars {
-                        expression.push(next);
-                        if next == ch {
-                            string_ended = true;
-                            break;
+                    loop {
+                        match self.next_char() {
+                            Some(c) => {
+                                expression.push(c);
+                                if c == ch {
+                                    break;
+                                }
+                            }
+                            None => return Err(UnterminatedString),
                         }
```

Comment:
```suggestion
                        let Some(c) = self.next_char() else {
                            return Err(UnterminatedString);
                        };
                        expression.push(c);
                        if c == ch {
                            break;
                        }
```

Reply:
I keep forgetting that let-else is a thing now.

Reply:
@burrbull Thanks for the suggestion!